deploy/helm-chart/templates/rbac.yaml

```diff
@@ -24,6 +24,7 @@ rules:
   # FIXME(bjee19): make nodes permission dependent on telemetry being enabled.
   # https://github.com/nginxinc/nginx-gateway-fabric/issues/1317.
   - nodes
+  - pods
```

Comment:
I tried not watching pods and I received an error message in the logs:

```
E0209 18:00:32.565150       7 reflector.go:147] k8s.io/client-go@v0.29.1/tools/cache/reflector.go:229: Failed to watch *v1.Pod: unknown (get pods)
E0209 18:00:32.645375       7 reflector.go:147] k8s.io/client-go@v0.29.1/tools/cache/reflector.go:229: Failed to watch *v1.ReplicaSet: unknown (get replicasets.apps)
E0209 18:00:36.007893       7 reflector.go:147] k8s.io/client-go@v0.29.1/tools/cache/reflector.go:229: Failed to watch *v1.Pod: unknown (get pods)
E0209 18:00:37.267612       7 reflector.go:147] k8s.io/client-go@v0.29.1/tools/cache/reflector.go:229: Failed to watch *v1.ReplicaSet: unknown (get replicasets.apps)
E0209 18:00:45.206742       7 reflector.go:147] k8s.io/client-go@v0.29.1/tools/cache/reflector.go:229: Failed to watch *v1.Pod: unknown (get pods)
E0209 18:00:46.098432       7 reflector.go:147] k8s.io/client-go@v0.29.1/tools/cache/reflector.go:229: Failed to watch *v1.ReplicaSet: unknown (get replicasets.apps)
```

Reply:
this is a concern. Now NGF watches for all Pods and Replicaset in the cluster - which is extra overhead for NGF and also Kubernetes API server.

The reason is that in manager go https://github.com/nginxinc/nginx-gateway-fabric/blob/6adf6ac115434dc3288667ec6c7c059a54d1cc6f/internal/mode/static/manager.go#L217 we path mgr.GetClient, which returns a client that Caches resources. 

If the resource doesn't have an Watcher (an Informer) for the corresponding resource type, it will create it -- https://github.com/kubernetes-sigs/controller-runtime/blob/bbe3bbe2fb38b76d87974b43ea5d3a750a319bc0/pkg/cache/internal/informers.go#L286

Using `mgr.GetAPIReader()` should solve this problem, as in this case, no caching will be involved. Instead, NGF will make a direct call to the API.

Note: we should also fix the RBAC for already implemented nodes

Reply:
Fixed, this let us change our HTTP request verbs to `get` (it did not work with list`).